sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregatesSuite.scala

```diff
@@ -96,7 +96,7 @@ class RemoveRedundantAggregatesSuite extends PlanTest {
         .groupBy('a + 'b)(('a + 'b) as 'c)
         .analyze
       val optimized = Optimize.execute(query)
-      comparePlans(optimized, expected)
+      comparePlans(optimized, EnforceGroupingReferencesInAggregates(expected))
```

Comment:
hmm, the `Optimize` in this test doesn't include the new rule `EnforceGroupingReferencesInAggregates`, why would the `optimzed` plan contains grouping refs?

Reply:
`RemoveRedundantAggregates` explicitly calls `Aggregate.withGroupingRefs()`: https://github.com/apache/spark/pull/31913/files#diff-11264d807efa58054cca2d220aae8fba644ee0f0f2a4722c46d52828394846efR511. Actually this call wouldn't be necessary if a `GroupingExprRef` could replace an `Attribute`...